iota-client/extended/prepare_transfers.rs

```diff
@@ -87,6 +87,16 @@ impl<'a> Client<'a> {
                 );
             }
             transfer.address = iota_signing::checksum::remove_checksum(&transfer.address);
+            if transfer.value > 0 {
+                ensure!(
+                    if transfer.address.trits()[242] == 0 {
+                        true
+                    } else {
+                        false
+                    },
```

Comment:
You don't need to have these since `transfer.address.trits()[242] == 0` is already a expression. Could you also only convert last tryte to trit, so we don't have to convert whole address?

Reply:
I don't know how to remove this part, if I just remove it I get
```
mismatched types

expected bool, found ()

note: expected type `bool`
         found type `()
````


To only convert the last char is it okay to do it this way? 
```Rust
let last_char = &transfer.address[transfer.address.len()-1..].to_string();
if last_char.trits()[2] == 0

Reply:
Maybe you still have the `if` keyword, just leave expression like this should work:
`ensure!(transfer.address.trits()[242] == 0,"Invalid Kerl address.");`
Regarding last tryte, I think let's just remain this way since `String` is not able to index. It's more about the design of type address. I will look into this when common crate rewrite.

Reply:
Thanks!